### DISCUSSION TITLE
data provider filter by and/or

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#3012] [Help] [answered]
Hello,

I would like to put a input box to allow user to filter rows by multiple fields, any of  rows fields matached the keyword will be returned. The SQL something like below. I don't want to add multiple form fields to filter it. Is there any workaround to do this?

`select * from tbl_customers where customerid like '%keyword%' or customername like  '%keyword%'  or email  like '%keyword%' `

<img width="1395" alt="iShot2021-10-14 23 14 34" src="https://user-images.githubusercontent.com/29649651/137346839-f210d761-a9db-4349-9226-a40343dfa325.png">

**Comment by mjashanks** [comment, marked as answer]
Hey @ggs331. Currently our filter only applies `and` - which is not ideal. This is something that we want to address asap.

In the meantime, since you're using SQL, you can write a custom query with parameters. These docs shows how parameters are used in custom queries. Although this uses our REST API connector, the usage of parameters is exactly the same in the SQL custom query.

https://docs.budibase.com/quickstart-tutorials/crud-app-with-a-rest-api/add-a-create-item-request

**Reply by samg666** [reply]
@mjashanks thanks for the reply. it seems the SQL Perameters is different from REST. There is no way to set the path. As I am new to budibase, can you share more details on how can I pass a form textbox value to the SQL as Perameters?

BTW, how can I reference a components value on a page? For example, I have a form1 datepicker1 value is 1 October 2021, I would like to set another datepicker2 default value = datapicker1. I tried to use something like {{form1.datapicker1}} on datapicker2 default value, but seems not work.

```
6.
Set the “Path” to /repos/<github_user>/<github-repo>/issues
```

**Reply by mjashanks** [reply]
hey @ggs331 - I've written this up in docs. I've been meaning to do this anyway. Let me know of any feedback!
https://docs.budibase.com/guides/writing-a-custom-sql-query

**Reply by samg666** [reply]
@mjashanks thank you very much, it works great.
